pkgs/by-name/ve/vesktop/package.nix

```
@@ -0,0 +1,62 @@
+{ buildNpmPackage, fetchFromGitHub, vencord, electron, pipewire, lib }:
+let
+  libPath = lib.makeLibraryPath [ pipewire ];
+in
+buildNpmPackage rec {
+  pname = "vesktop";
+  version = "0.3.3";
+
+  src = fetchFromGitHub {
+    owner = "Vencord";
+    repo = "Vesktop";
+    rev = "v${version}";
+    sha256 = "sha256-Njs3tACxUyRolYUtS/q2lITIQnUBFXVXWZEfQ66HpPM=";
+  };
+
+  prePatch = ''
+    cat ${./package-lock.json} > ./package-lock.json
+  '';
+
+  patches = [
+    ./use_system_vencord.patch
+  ];
+
+  postPatch = ''
+    sed -i "s!@VENCORD_STORE_DIR@!${vencord}!" src/main/constants.ts
+  '';
+
+  npmDepsHash = "sha256-X4/7N+l92CfTfD3XXifRd5G0oxGWvIUl8CG+foAyShc=";
+  npmFlags = [ "--legacy-peer-deps" ];
+  makeCacheWritable = true;
+  env.ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+
+  # this is consistent with other nixpkgs electron packages and upstream, as far as I am aware
+  # yes, upstream really packages it as "vesktop" but uses "vencorddesktop" file names
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/lib/node_modules/VencordDesktop
+    cp -r dist/ $out/lib/node_modules/VencordDesktop
+    cp -r static/ $out/lib/node_modules/VencordDesktop
+    cp package.json $out/lib/node_modules/VencordDesktop
+
+    install -D static/icon.png $out/share/icons/hicolor/scalable/apps/vencorddesktop.png
+    install -Dm444 ${./vencorddesktop.desktop} $out/share/applications/vencorddesktop.desktop
+
+    makeWrapper ${electron}/bin/electron $out/bin/vencorddesktop \
+      --prefix LD_LIBRARY_PATH : ${libPath} \
+      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform=wayland}}" \
+      --add-flags $out/lib/node_modules/VencordDesktop
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "An alternate client for Discord with Vencord built-in";
+    homepage = "https://github.com/Vencord/Vesktop";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ vgskye ];
```

I would like to co-maintain this package :O

```suggestion
    maintainers = with maintainers; [ Scrumplex vgskye ];
```